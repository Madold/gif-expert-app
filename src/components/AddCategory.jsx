import { useState } from "react"
import PropTypes from "prop-types"

export const AddCategory = ({ onNewCategory }) => {

    const [categoryName, setCategoryName] = useState("")

    const onValueChange = ({ target }) => setCategoryName(target.value)

    const handleOnSubmit = e => {
        e.preventDefault()
        if (categoryName.trim() !== "") {
            onNewCategory(categoryName)
            resetValue()
        }
    }

    const resetValue = () => setCategoryName("")

    return (
        <form onSubmit={handleOnSubmit} aria-label="form" >
            <input
                type="text"
                placeholder="Buscar gif"
                value={categoryName}
                onChange={onValueChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
} 