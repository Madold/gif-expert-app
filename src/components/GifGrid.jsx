import { useFetchGifs }  from "../hooks/useFetchGifs"
import { LoadingIndicator, GifGridItem }  from "./"
import PropTypes from "prop-types"


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>

            {
                isLoading && <LoadingIndicator />
            }

            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img} />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}