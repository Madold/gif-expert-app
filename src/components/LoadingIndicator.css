.loading-indicator {
    width: 120px;
    height: 120px;
    border: 2px solid transparent;
    border-left-color: #3498db;
    border-radius: 100%;
    animation: spin 1s linear infinite;
}

@keyframes spin {
    to {
        transform: rotate(360deg);
    }
}