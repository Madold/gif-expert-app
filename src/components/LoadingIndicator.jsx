//import "./LoadingIndicator.css"


export const LoadingIndicator = () => { 

    return (
        <div className="loading-indicator" aria-label="loading-indicator"></div>
    )
}
