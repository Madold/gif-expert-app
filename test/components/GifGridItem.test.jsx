import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";


describe('Pruebas en GifGridItem', () => {

    const title = "Saitama"
    const url = "https://localhost/algo.jpg"


    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL indicado', () => {
        render(<GifGridItem title={title} url={url} />)
        //screen.debug()
        const { src } =  screen.getByRole("img")
        expect(src).toBe(url)
    });    

});