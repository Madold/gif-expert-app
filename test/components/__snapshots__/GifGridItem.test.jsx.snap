// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://localhost/algo.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
